tests/tasks/setup/eks/awscli-cp.yaml

```diff
@@ -0,0 +1,39 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: awscli-eks-cluster-create
+spec:
+  description: |
+    Create an EKS cluster.
+    This Task can be used to create an EKS cluster for a given VPC Subnets, security groups and service role in an AWS account and write a kubeconfig to a desired location that
+    can be used by other tasks (in a context with kubectl) to make requests to the cluster.
+  params:
+  - name: cluster-name
+    description: The name of the EKS cluster you want to spin.
+  - name: kubernetes-version
+    default: "1.21"
+    description: The EKS version to install.
+  - name: region
+    default: "us-west-2"
+    description: The region where the cluster is in.
+  - name: endpoint
+    default: "https://api.wesley.amazonaws.com"
+  - name: subnets
+    description: list of subnet ids of the VPC for eks cluster.
+  - name: securitygroups
+    description: list of securitygroup ids as a string of the associated VPC
+  - name: servicerole
+    description: servicerole arn to be used for eksc cluster
+  workspaces:
+  - name: config
+    description: |
+      A workspace into which a kubeconfig file called `kubeconfig` will be written that will contain the information required to access the cluster. The `kubeconfig` will expect to use [aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator/) to authenticate, so in order for it to be used it must be run in a container which contains both `kubectl` and `aws-iam-authenticator`.
+  steps:
+  - name: write-kubeconfig
+    image: amazon/aws-cli
+    script: |
+      aws eks create-cluster --name $(params.cluster-name) --kubernetes-version $(params.kubernetes-version) --role-arn $(params.servicerole) --resources-vpc-config subnetIds=$(params.subnets),securityGroupIds=$(params.securitygroups) --endpoint $(params.endpoint)
```

Comment:
Can we default the endpoint to empty and only add `--endpoint $(params.endpoint)` when a non-empty endpoint URL is given?